c++/src/capnp/rpc.c++

```diff
@@ -503,7 +503,8 @@ public:
     auto shutdownPromise = dyingConnection->shutdown()
         .attach(kj::mv(dyingConnection))
         .then([]() -> kj::Promise<void> { return kj::READY_NOW; },
-              [this, origException = kj::mv(exception)](kj::Exception&& shutdownException) -> kj::Promise<void> {
+              [self = kj::addRef(*this), origException = kj::mv(exception)](
```

Comment:
Here I'm assuming it's OK to capture an additional strong reference to the RpcConnectionState in the error handler -- i.e. that doing so doesn't cause a reference cycle that would cause objects to leak... Not sure if there is existing machinery to detect such problems.

If we want something less likely to form cycles (but also less performant), we could move the receiveIncomingMessageError flag into its own refcounted object and refer to it from both RpcConnectionState and this lambda.

Reply:
This looks correct -- the promise being fulfilled is not itself held by the RpcConnectionState so there's no cycle.